lib/sift/value_parser.rb

```diff
@@ -39,6 +40,8 @@ def parse_as_range?
     end
 
     def range_value
+      @value = normalized_date_range if [:date, :time, :datetime].include?(type)
+
```

Comment:
Continuing the conversation here. 

I think it's risky to modify @value here. If @value should be modified, it would be better to do in the initializer. This can cause an order dependence and side affect to calling `parse`. parse would actually modify the result of @value (which is exposed in the public method `.value`)